Classes/ios/UIViewController+MBDebugPanel.m

```diff
@@ -0,0 +1,157 @@
+//
+//  UIViewController+MBDebugPanel.m
+//  MBDebugPanel
+//
+//  Created by Ian on 3/27/15.
+//
+//
+
+#import "UIViewController+MBDebugPanel.h"
+
+#import "MBDebugPanel.h"
+
+#import <JGMethodSwizzler/JGMethodSwizzler.h>
+
+#import <objc/runtime.h>
+
+NSString *const MB_UIViewController_Utility_debugMenuItemsManagementEnabledKey = @"MB_UIViewController_Utility_debugMenuItemsManagementEnabled";
+NSString *const MB_UIViewController_Utility_managedDebugMenuItemsKey = @"MB_UIViewController_Utility_managedDebugMenuItemsKey";
+
+@interface UIViewController (MBDebugPanel_Private)
+
+@property (retain, readwrite) NSMutableArray *mb_managedDebugMenuItems;
+
+@end
+
+@implementation UIViewController (MBDebugPanel)
+
+- (void)setMb_debugMenuItemsManagementEnabled:(BOOL)mb_debugMenuItemsManagementEnabled
+{
+    // TODO: test what happens if we call this from viewWillAppear or viewDidDisappear
+
+    if (mb_debugMenuItemsManagementEnabled != self.mb_debugMenuItemsManagementEnabled)
+    {
+        if (mb_debugMenuItemsManagementEnabled == true)
+        {
+            [self swizzleMethod:@selector(viewWillAppear:) withReplacement:JGMethodReplacementProviderBlock {
+                return JGMethodReplacement(void, UIViewController *, BOOL animated) {
+                    JGOriginalImplementation(BOOL, animated);
+                    
+                    [MBDebugPanel addComponentsFromArray:self.mb_managedDebugMenuItems];
+                };
+            }];
+            
+            [self swizzleMethod:@selector(viewDidDisappear:) withReplacement:JGMethodReplacementProviderBlock {
```

Comment:
I chose WillAppear and DidDisappear because they are both outside of the range of possible visibility to the user, granted this is overly safe and probably not really necessary.